### PR TITLE
Queue is fertig

### PR DESCRIPTION
**Pull request by Daki21**
pls review code

### CODE REVIEW

**Review by KuroiX — CHANGES_REQUESTED** [review]
Needs improvement

File: Assets/CustomQueue.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+public class CustomQueue<T> : MonoBehaviour
```
[the remark this answers is not in the document]
Comment (Daki21) [reply Assets/CustomQueue.cs:6]: why tho

[the remark this answers is not in the document]
Comment (KuroiX) [reply Assets/CustomQueue.cs:6]: MonoBehavior is a class provided by unity so you can attack it as a component to a GameObject. The queue however is not supposed to be attachable.

[the remark this answers is not in the document]
Comment (Daki21) [reply Assets/CustomQueue.cs:6]: but why do you want to attack it :O

**Comment by KuroiX** [comment]
I dont know how to re-review the code D:

File: Assets/CustomQueue.cs [hunk 2]
```diff
@@ -1,55 +1,46 @@
...
-using UnityEngine;
 
-public class CustomQueue<T> : MonoBehaviour
-{
-    private List<T> queue = new List<T>();
-    private int counter = 0;
-    private int queueSize = 7;
-
-    public bool enqueue(T value)
+public class CustomQueue<T> {
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:3]: Description of class is missing (same as with methods)

File: Assets/CustomQueue.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        _list = new List<T>();
+        _index = 0;
+    }
+
+    /// <summary>
+    /// Queue up a T
+    /// </summary>
+    /// <param name="value">This is the parameter that gets queued up</param>
+    /// <returns>true if successful, false if not successful</returns>
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:18]: Summary should answer the question "What does this method/class do?" so the answer would be in third person singular: "Queues up a T"
Also: "queue up" not coherent with "enqueue, t not very specific -> "Enqueues an item"
"this is the parameter that gets queued up" -> "Item that gets enqueued"
<returns> does not exist anymore in the new method

File: Assets/CustomQueue.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    public void Enqueue(T value)
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:19]: rename "value" to "item"

File: Assets/CustomQueue.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        _list.Add(value);
+    }
+
+    /// <summary>
+    /// Dequeue a T
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:25]: Dequeues the next item

File: Assets/CustomQueue.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    /// </summary>
+    /// <returns>T that got dequeued</returns>
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:27]: Item that got dequeued

File: Assets/CustomQueue.cs [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    public T Dequeue()        
+    {
+         if (IsEmpty())
+            return default;
+        
+         return _list[_index++];
+    }
+
+    /// <summary>
+    /// See the next T to be dequeued
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:37]: answer "what does the method do?"

File: Assets/CustomQueue.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    /// </summary>
+    /// <returns>the next T to be dequeued</returns>
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:39]: The

File: Assets/CustomQueue.cs [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    public T Peek()
+    {
+        return _list[_index];
+    }
+
+    /// <summary>
+    /// Tells you if the queue is empty
+    /// </summary>
+    /// <returns>true if empty, false if not empty</returns>
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:48]: True

File: Assets/CustomQueue.cs [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    public bool IsEmpty()
+    {
+        return _index >= _list.Count;
+    }
+
+    /// <summary>
+    /// Reset the queue
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:55]: Resets the queue

File: Assets/CustomQueue.cs [hunk 11]
```diff
@@ -0,0 +1,61 @@
+using System.Collections.Generic;
+
+public class CustomQueue<T> {
```
Comment (KuroiX) [inline Assets/CustomQueue.cs:3]: Rename to "ResettableQueue" for better expressivness

File: Assets/CustomQueue.cs (hunk 4 above)
Comment (KuroiX) [reply Assets/CustomQueue.cs:19]: keep in mind when changing the method head to also change the description, e.g. the parameter name

**Review by KuroiX — CHANGES_REQUESTED** [review]
Good changes, i could not find any mistakes. The comments can be improved to be more descriptive.

Has the queue been tested?

**Comment by KuroiX** [comment]
Test

**Review by KuroiX — APPROVED** [review]

**Comment by Daki21** [comment]
Hey ich bin fertig

**Pull request merged**